### PR TITLE
clientupdate: parse /etc/synoinfo.conf to get CPU arch

### PR DESCRIPTION
**Pull request by awly**
The hardware version in `/proc/sys/kernel/syno_hw_version` does not map exactly to versions in
https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures. It contains some slightly different version formats.

Instead, `/etc/synoinfo.conf` exists and contains a `unique` line with the CPU architecture encoded. Parse that out and filter through the list of architectures that we have SPKs for.

Tested on DS218 and DS413j.

Updates #8927

### CODE REVIEW

File: clientupdate/clientupdate_test.go [hunk 1]
```diff
@@ -446,29 +444,108 @@ tailscale installed size:
...
+			content: `
+company_title="Synology"
+`,
+			wantErr: true,
+		},
+		{
+			desc: "empty unique",
+			content: `
+company_title="Synology"
+unique=""
```
Comment (noncombatant) [inline clientupdate/clientupdate_test.go:529]: Maybe also try blank and without quotes. (Should we handle single quotes, too?)

File: clientupdate/clientupdate_test.go [hunk 2]
```diff
@@ -446,29 +444,108 @@ tailscale installed size:
...
-		{goarch: "arm", model: "DS115", want: "armada375"},
-		{goarch: "arm", model: "DS419slim", want: "armada38x"},
-		{goarch: "arm", model: "RS815", want: "armadaxp"},
-		{goarch: "arm", model: "DS414j", want: "comcerto2k"},
-		{goarch: "arm", model: "DS216play", want: "monaco"},
-		{goarch: "riscv64", model: "DS999", wantErr: true},
+		{
+			desc: "normal",
+			content: `
+company_title="Synology"
```
Comment (noncombatant) [inline clientupdate/clientupdate_test.go:505]: Do we have an example of a complete file? (Or is this it?)

File: clientupdate/clientupdate_test.go (hunk 1 above)
Comment (awly) [reply clientupdate/clientupdate_test.go:545]: Added more test cases!

File: clientupdate/clientupdate_test.go (hunk 2 above)
Comment (awly) [reply clientupdate/clientupdate_test.go:505]: The complete file is like 380 lines long.
It's a kitchen sink of undocumented parameters :(

Comment (noncombatant) [reply clientupdate/clientupdate_test.go:505]: Wow! Alas.

**Review by noncombatant — APPROVED** [review]

**Pull request merged**